peer/peer.go

```diff
@@ -1670,6 +1670,32 @@ cleanup:
 	log.Trace("Peer queue handler done for %s", p)
 }
 
+func (p *Peer) PeerHasHeader(idx *blockindex.BlockIndex) bool {
+	gChain := chain.GetInstance()
+	hash := p.LastAnnouncedBlock()
+	if hash == nil {
+		genesishash := gChain.GetParams().GenesisBlock.GetHash()
+		hash = &genesishash
+		p.UpdateLastAnnouncedBlock(hash)
+	}
+	pindex := gChain.FindBlockIndex(*hash)
+	if pindex == nil {
+		return false
```

Comment:
too early return, should also check IndexBestHeaderSent before return false.

Reply:
it's really a bug, I will fix it right now.